source/includes/inkling2-reference/_curriculums.html.md

```diff
@@ -113,8 +113,123 @@ graph (input: GameState) {
 
+### STAR in Inkling
+When using a simulator as the source of training data, each training iteration involves an input to the simulator (an “action”) and an output from the simulator (the new “state”). In addition, a “reward” value is generated along with a “terminal” flag that indicates whether the training episode should be ended. These four piece of data (state, terminal, action, and reward) are sometimes collectively referred to as “STAR”.
+
+#### Reward Functions
+While it’s possible for the simulator code to compute the reward value, it’s often more convenient to implement this reward function in Inkling code. This allows for more rapid experimentation. Simple reward functions can be implemented using the [function][6] capabilities in Inkling.
+
+If a reward function is specified in Inkling, the reward value passed from the simulator is ignored.
+
+An Inkling reward function takes a single input parameter — the new state that was returned from the simulator. It must return a numeric value indicating the reward associated with that state. The reward function is specified within the curriculum statement using the `reward` keyword.
+
+```inkling2--code
+concept Balance(input: SimState): Action {
+    curriculum {
+        source MySimulator
+        reward GetReward
+    }
+}
+
+function IsUpright(Angle: number) {
+  return Angle > 85 and Angle < 95
+}
+
+function GetReward(State: SimState) {
+  if IsUpright(State.Angle) {
+      return 1.0
+  }
+  return -0.01
+}
+```
+
+#### Terminal Functions
+As with reward functions, terminal functions can be written within Inkling. If a terminal function is specified, the terminal value passed from the simulator is ignored.
+
+An Inkling terminal function takes a single input parameter — the new state that was returned from the simulator. It must return true (1) if the state is a terminal state or false (0) otherwise. The terminal function is specified within the curriculum statement using the `terminal` keyword.
+
+Note that a function is specified inline in the provided example code.
+
+```inkling2--code
+concept Balance(input: SimState): Action {
+    curriculum {
+        source MySimulator
+        terminal function (State: SimState) {
+          if State.IterationCount > 100 {
+            return true
+          }
+          return State.Position > 20
+        }
+    }
+}
+```
+
+#### State Transform Functions
+The “state” data returned by a simulator represents the state of the simulated environment, including all observables. In many cases, the simulator is able to produce more observable values than are available in the real world. In such cases, it’s important to train the BRAIN using only the subset of real-world observables. It’s important to distinguish between the “simulator state” and the “training state”.
```

Comment:
terminology note: the CS template and various SDK3 discussions have been calling these "observations" and "state" or "training state" -- I've found that easy to explain to people, so suggest we use "observations" instead of "sim state".